server/remote/remote.go

```diff
@@ -10,11 +10,12 @@ import (
 	"golang.org/x/oauth2"
 
 	"github.com/mattermost/mattermost-plugin-mscalendar/server/config"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/store"
 	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/bot"
 )
 
 type Remote interface {
-	MakeClient(context.Context, *oauth2.Token) Client
+	MakeClient(context.Context, bot.Poster, store.Store, string, *oauth2.Token) Client
```

Comment:
```suggestion
	MakeClient(context.Context, bot.Poster, store.Store, mattermostUserID string, *oauth2.Token) Client
```

Reply:
can not give mix of named and unnamed parameters